File: lambdat/raportointi/src/main/scala/fi/oph/viestinvalitys/raportointi/configuration/SessionMappingStorageConfiguration.scala
package fi.oph.viestinvalitys.raportointi.configuration;

import fi.oph.viestinvalitys.raportointi.security.JdbcSessionMappingStorage
import org.apereo.cas.client.session.SessionMappingStorage
import org.slf4j.LoggerFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.session.{Session, SessionRepository}
import org.springframework.session.jdbc.JdbcIndexedSessionRepository;

@Configuration
class SessionMappingStorageConfiguration {
    val LOG = LoggerFactory.getLogger(classOf[SessionMappingStorageConfiguration])
    @Bean
    def sessionMappingStorage(sessionRepository: JdbcIndexedSessionRepository): SessionMappingStorage = {
        val jdbcSessionMappingStorage = new JdbcSessionMappingStorage(sessionRepository.asInstanceOf[SessionRepository[Session]]);
        jdbcSessionMappingStorage
    }
}

File: viestinvalitys-service/src/main/java/fi/vm/sade/viestinvalitys/config/SessionMappingStorageConfiguration.java
package fi.vm.sade.viestinvalitys.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.session.jdbc.JdbcIndexedSessionRepository;

/**
 * Exposes the CAS single-logout {@link OphSessionMappingStorage} as a bean so it can be injected
 * into {@code SecurityConfig}'s SingleSignOutFilter. Mirrors the equivalent configuration in
 * oppijanumerorekisteri-service / kayttooikeus-service.
 */
@Configuration
public class SessionMappingStorageConfiguration {

    @Bean
    OphSessionMappingStorage sessionMappingStorage(
            JdbcTemplate jdbcTemplate, JdbcIndexedSessionRepository sessionRepository) {
        return new JdbcSessionMappingStorage(jdbcTemplate, sessionRepository);
    }
}
